src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
                 }
 
                 return default;
-            });
-    }
+            };
 
-    /// <summary>
-    /// Static members for <see cref="Driver{TCallable,TIn,TOut}"/>.
-    /// </summary>
-    internal static class Driver
-    {
-        // TODO: Define the aliases as constants.
+            command.AddValidator(validator);
+            return new[] { validator };
+        }
 
         /// <summary>
-        /// The subscription option.
+        /// Creates the simulate command.
         /// </summary>
-        internal static readonly OptionInfo<string> SubscriptionOption = new OptionInfo<string>(
-            ImmutableList.Create("--subscription"), "The subscription ID.");
+        /// <param name="entryPointCommands">The entry point commands that will be the sub commands to the created command.</param>
+        /// <returns>Tuple with the created simulate command and the validators for that command.</returns>
+        private static (Command, IEnumerable<ValidateSymbol<CommandResult>>) CreateSimulateCommand(
+            IEnumerable<(Command, IEnumerable<ValidateSymbol<CommandResult>>)> entryPointCommands)
+        {
+            var validators = Enumerable.Empty<ValidateSymbol<CommandResult>>();
 
-        /// <summary>
-        /// The resource group option.
-        /// </summary>
-        internal static readonly OptionInfo<string> ResourceGroupOption = new OptionInfo<string>(
-            ImmutableList.Create("--resource-group"), "The resource group name.");
+            var simulate = new Command("simulate", "(default) Run the program using a local simulator.");
+            if (entryPointCommands.Count() == 1)
+            {
+                var (epCommand, valids) = entryPointCommands.First();
+                simulate.AddCommand(epCommand);
+                SetSubCommandAsDefault(simulate, epCommand, valids);
+                validators = valids;
```

Comment:
nit: Same as above - you can return from both branches instead of using a mutable variable.

Reply:
done